### DISCUSSION TITLE
How to debug performance issues with Nextcloud?

### DISCUSSION THREAD

**Discussion by szaimen** [nextcloud/all-in-one#2475] [Wiki]
It can happen that you face performance issues with Nextcloud, like having to wait something like 60 or more seconds until the Nextcloud web interface loads. See the steps below how to debug this. Additionally see https://github.com/nextcloud/all-in-one/discussions/1335 for general recommendations.

A first step would be to temporarily disable Nextclouds bruteforce protection and ratelimit protection by running `sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ config:system:set auth.bruteforce.protection.enabled --type bool --value=false`
and `sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ config:system:set ratelimit.protection.enabled --type bool --value=false` and check if it fixes the performance problems.

If that should be the case, you are likely facing problems due to Nextclouds own bruteforce protection or ratelimit protection limiting the requests.

Since the above setting is obviously not secure, you should next enable the protection again by running `sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ config:system:set auth.bruteforce.protection.enabled --type bool --value=true`
and `sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ config:system:set ratelimit.protection.enabled --type bool --value=true`

Afterwards, try to log in with a random account and random password into Nextcloud from an icognito tab from the same device and check in the server logs `https://yourdomain.com/settings/admin/logging` which ip-address gets logged (usually the public ipv4-address or ipv6-address of your device). If it should be an unexpected private ip-address, you have likely found the culprit. See https://github.com/nextcloud/all-in-one/discussions/2045 how to proceed. 

Afterwards, the correct ip-address should usually be logged and the ratelimiting and bruteforce protection only happen for requests that actually trigger it. If it should still be slow without disabling the bruteforce and ratelimit protection, feel free to create a new discussion here: https://github.com/nextcloud/all-in-one/discussions/new?category=questions

**Comment by mitexleo** [comment]
Is there any way to load NextCloud under 1 second? I'm using Hetzner's CX32 VPS.

**Reply by mitexleo** [reply]
I disabled rate limiting and the Web UI is a lot faster now. I don't the trusted_proxy issue you've mentioned (already fixed it before).

**Reply by szaimen** [reply]
The trusted proxy setting is the correct way. Disabling rate limiting is an incorrect workaround and has security implications

**Reply by mitexleo** [reply]
Log shows my correct IP.  I also have Bruteforce protection enabled. Is there any documentation on how the rate limit works? I'm the only user on my server. 

Here's my trusted_proxies:

![SmartSelect_20241022-201348_Termux](https://github.com/user-attachments/assets/dd6d1dfa-d4a2-47b0-b5c1-96023a09af88)

**Reply by szaimen** [reply]
Please open a dedicated discussion for this. Thanks!
